### DISCUSSION TITLE
List of all the states

### DISCUSSION THREAD

**Discussion by jlpouffier** [teslamate-org/teslamate#3049] [Getting started] [answered]
Hello,

I am searching for a list of all the states that can be returned on the mqtt topic `teslamate/cars/$car_id/state`

Some example includes:

- online
- asleep
- etc

The docs ([here](https://docs.teslamate.org/docs/integrations/mqtt/)) only mentions 3 exmaples.

I am looking for this list to be able to translate these status nicely on my home automation platform.

Thx !

**Comment by cwanja** [comment, marked as answer]
Here is the legend from the `States` Grafana dashboard. 
![Screen Shot 2023-01-09 at 7 18 09 AM](https://user-images.githubusercontent.com/54645561/211317113-bd6eeeb0-586f-4137-b3e0-4f73958ae174.png)

**Reply by jlpouffier** [reply]
thx !
Checking grafana was smart :)

**Reply by jlpouffier** [reply]
I just marked your answer as "unanswered" because I just saw a state that is not part of this list on MQTT.
The value is `suspended`.

It happens when Teslamate decides to stop querying the car to avoid stopping the shutdown.

On Teslamate it looks like this: 
![image](https://user-images.githubusercontent.com/5878296/211345351-41619ca9-acaf-4f73-b997-b9a23acd6068.png)

And on my MQTT explorer, like this: 
![image](https://user-images.githubusercontent.com/5878296/211345401-c5c58d39-d1a1-4b7a-997f-b02d99b3cd51.png)
 
So if you have a better idea where I can find a "definitive list" of all the state that can be returned on the topic `teslamate/cars/$car_id/state` that would be aaamazing :) 

Thx !

**Reply by jlpouffier** [reply]
Maybe [this file](https://github.com/adriankumpf/teslamate/blob/6c537887381349b5171cdf577601883c2aaf7a28/lib/teslamate_web/live/car_live/summary.ex#L124) could be the answer?

![image](https://user-images.githubusercontent.com/5878296/211346331-4a72445b-242b-4876-a1da-f3e0d6410a04.png)

I see 8 states:
- start
- driving
- charging
- updating
- suspended
- online
- offline
- asleep
- unavailable

**Reply by cwanja** [reply]
> I just marked your answer as "unanswered" because I just saw a state that is not part of this list on MQTT.
> 
> The value is `suspended`.
> 
> 
> 
> It happens when Teslamate decides to stop querying the car to avoid stopping the shutdown.
> 
> 
> 
> On Teslamate it looks like this: 
> 
> ![image](https://user-images.githubusercontent.com/5878296/211345351-41619ca9-acaf-4f73-b997-b9a23acd6068.png)
> 
> 
> 
> And on my MQTT explorer, like this: 
> 
> ![image](https://user-images.githubusercontent.com/5878296/211345401-c5c58d39-d1a1-4b7a-997f-b02d99b3cd51.png)
> 
>  
> 
> So if you have a better idea where I can find a "definitive list" of all the state that can be returned on the topic `teslamate/cars/$car_id/state` that would be aaamazing :) 
> 
> 
> 
> Thx !

That is because `suspended` is not a state that the OEM API reports. It's a 'hybrid' state, so to speak, that TeslaMate is in trying to let the car fall asleep. That is why you do not see 'suspended' as a logged state in Grafana. Because your car is still 'online', but TeslaMate has suspended polling to let it fall asleep. And when the API reports asleep, TeslaMate will report your car as asleep.

**Reply by jlpouffier** [reply]
Ok ! Understood !

Then I got my answer :) 
Thx a bunch !

I translated everything I needed. 

I'll mark this discussion as done :)
